### PR TITLE
fix: Lottery you won card after claim

### LINKED ISSUE #828: Lottery "You won!" card does not disappear after collection rewards.
**Bug Description**
When you collect your lottery prize the "You won!" card does not go away after collection.

**Steps to Reproduce**

1. Go to lottery
2. Win the lottery
3. Collect your prize
4. After transaction is completed and you got your cake - you still see the "You won!" card. Clicking on "Collect" again does nothing.

**Expected Behavior**
After you collected your prize you "You won!" card should disappear (or change to something else).

### PR DESCRIPTION
**Pull request by memoyil**
Fixes #828

### CODE REVIEW

**Review by ChefHutch — COMMENTED** [review]
Do we have any way to test lottery wins on testnet @Chef-Chungus to validate that this fixes the issue?

File: src/views/Lottery/components/YourPrizesCard/PrizesWonContent.tsx [hunk 1]
```diff
@@ -52,12 +56,13 @@ const PrizesWonContent: React.FC = () => {
       const txHash = await onMultiClaim()
       // user rejected tx or didn't go thru
       if (txHash) {
+        afterClaim()
```
Comment (hachiojidev) [inline src/views/Lottery/components/YourPrizesCard/PrizesWonContent.tsx:59]: For success/fail handlers lets use a standard naming scheme. `onSuccess` and `handleSuccess`

File: src/views/Lottery/components/YourPrizesCard/index.tsx [hunk 2]
```diff
@@ -32,9 +32,16 @@ const YourPrizesCard: React.FC = () => {
   const winnings = getBalanceNumber(claimAmount)
   const isAWin = winnings > 0
 
+  const [, updateState] = useState<any>()
+  const forceUpdate = useCallback(() => updateState({}), [])
+
+  const handleAfterClaim = useCallback(() => {
+    forceUpdate()
+  }, [forceUpdate])
+
```
Comment (hachiojidev) [inline src/views/Lottery/components/YourPrizesCard/index.tsx:41]: I have used something similar elsewhere so I created a hook that will achieve the same result (along with some other use-cases)
https://github.com/pancakeswap/pancake-frontend/pull/876

File: src/views/Lottery/components/YourPrizesCard/index.tsx [hunk 3]
```diff
@@ -1,4 +1,4 @@
-import React from 'react'
+import React, { useCallback, useEffect, useState } from 'react'
```
Comment (hachiojidev) [inline src/views/Lottery/components/YourPrizesCard/index.tsx:1]: Can you remove unused variables.

**Review by hachiojidev — CHANGES_REQUESTED** [review]

File: src/views/Lottery/components/YourPrizesCard/index.tsx (hunk 3 above)
Comment (memoyil) [reply src/views/Lottery/components/YourPrizesCard/index.tsx]: Done

File: src/views/Lottery/components/YourPrizesCard/PrizesWonContent.tsx (hunk 1 above)
Comment (memoyil) [reply src/views/Lottery/components/YourPrizesCard/PrizesWonContent.tsx]: Done

File: src/views/Lottery/components/YourPrizesCard/index.tsx (hunk 2 above)
Comment (memoyil) [reply src/views/Lottery/components/YourPrizesCard/index.tsx]: Waiting for that pr to be merged

Comment (memoyil) [reply src/views/Lottery/components/YourPrizesCard/index.tsx]: Done

**Review by hachiojidev — APPROVED** [review]

**Pull request merged**